src/main/java/sana/Memory.java

```diff
@@ -0,0 +1,103 @@
+package sana;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.Scanner;
+
+import sana.task.Deadline;
+import sana.task.Event;
+import sana.task.Task;
+import sana.task.ToDo;
+
+
+/**
+ * The Memory class abstracts the leading and saving of tasks given to Sana
+ *
+ * @author Jan
+ * @version 1.0
+ */
+public class Memory {
+    /** This variable is the path where the taskList is saved */
+    private static final String MEM_PATH = "data/sana.txt";
+
+    /** This variable is the path for directory that stores the memory file */
+    private static final String DIR_PATH = "data";
+
+    /** This variable represents the File object that stores the taskList */
+    private File memFile;
+
+    /** The constructor for a Memory object */
+    public Memory() {
+        initialiseMem();
+    }
+
+    /** Create new sana.txt file and data dir in case of corruption */
+    private void initialiseMem() {
+        File dir = new File(DIR_PATH);
+        if (!dir.exists()) {
+            dir.mkdir();
+        }
+        memFile = new File(MEM_PATH);
+        if (!memFile.exists()) {
+            try {
+                memFile.createNewFile();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        this.memFile = new File(MEM_PATH);

```

Comment:
I noticed that the `this` keyword is not used when referring to `memFile` anywhere else in this function. Perhaps consider making this line consistent too?